automation/automation/DriverProgram.cs

```diff
@@ -0,0 +1,90 @@
+            driver.Navigate().Refresh();
+            //Waiting for the header to load
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("//body/div[@id='container']/div/div[2]/div[1]")));
+            
+            //Adding 1st item in the wishlist
+            IWebElement electronics = driver.FindElement(By.XPath("//span[contains(text(),'Electronics')]"));
+            Thread.Sleep(2000);            
+            electronics.Click();        
+            IWebElement eleVivo = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//a[contains(text(),'Vivo')]")));
+            // clicking for product's page
+            eleVivo.Click();
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[1]")));
+            IWebElement addToWishList1 = driver.FindElement(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[2]"));
+            addToWishList1.Click();
+
+            //Adding item 2nd in the wishlist
+            IWebElement eleMen = driver.FindElement(By.XPath("//span[contains(text(),'Men')]"));
+            Thread.Sleep(2000);
+            eleMen.Click();
+            IWebElement eleTrou = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//a[contains(text(),'Casual Trousers')]")));
+            // clicking for product's page
+            eleTrou.Click();
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[3]")));
+            IWebElement addToWishList2 = driver.FindElement(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[3]"));
+            addToWishList2.Click();
+
+            //Adding item 3rd in the wishlist
+            IWebElement eleBabyKids = driver.FindElement(By.XPath("//span[contains(text(),'Baby & Kids')]"));
+            Thread.Sleep(2000);
+            eleBabyKids.Click();
+            IWebElement eleSoftToys = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//a[contains(text(),'Soft Toys')]")));
+            // clicking for product's page
+            eleSoftToys.Click();
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[3]")));
+            IWebElement addToWishList3 = driver.FindElement(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[2]"));
+            addToWishList3.Click();
+
+            //Moving on to the user and checking the wishlisted elements count
+            IWebElement user = driver.FindElement(By.XPath("//body/div/div/div/div/div/div[3]/div[1]"));
```

Comment:
this kind of xpath is usually considered unacceptable simply due to the number of tags involved. makes the entire identifier prone to failure.
You can explore xpaths like below that use a neighboring element that has a better attribute we can use, and then navigate to the required element.

Here, we start from the search bar which has a good title value we can use, and then navigate to the adjacent user button.
//input[contains(@title, 'Search for products')]/ancestor::form/../following-sibling::div

Reply:
I have simplified the xpath